File: 0389-find-the-difference/0389-find-the-difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        // t elements are randomly shuffled so we need to sort them
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());
        int i=0,j=0;
        while(i<s.length() && j<t.length()){
             if(s[i]!=t[j]) return t[j];
             i++;
             j++;
        }
        return t[j];
    }
};

File: 0389-find-the-difference/0389-find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        // t elements are randomly shuffled so we need to sort them
        char[] s1= s.toCharArray();
        char[] t1=t.toCharArray();
        Arrays.sort(s1);
        Arrays.sort(t1);
        int i=0,j=0;
        while(i<s1.length && j<t1.length){
             if(s1[i]!=t1[j]) return t1[j];
             i++;
             j++;
        }
        return t1[j];
    }
}